apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -155,11 +175,75 @@ public override void ActivateOptions()
+                // and platform detected is not prefered, then just use the manual configuration.
+                _logName = new LogName(ProjectId, LogId).ToString();
+                _resource = new MonitoredResource
+                {
+                    Type = ResourceType,
+                    Labels = { _resourceLabels.ToDictionary(x => x.Key, x => x.Value) }
+                };
+            }
+            else
+            {
+                // Use auto-detected platform information.
+                string platformProjectId;
+                MonitoredResource platformResource;
+                var platform = Platform.Instance();
+                switch (platform.Type)
+                {
+                    case PlatformType.Gae:
+                        var gae = platform.GaeDetails;
+                        platformProjectId = gae.ProjectId;
+                        platformResource = new MonitoredResource
+                        {
+                            Type = "gae_app",
+                            Labels =
+                            {
+                                { "project_id", gae.ProjectId },
+                                { "module_id", gae.ServiceId },
+                                { "version_id", gae.VersionId }
+                            }
+                        };
+                        break;
+                    case PlatformType.Gce:
+                        var gce = platform.GceDetails;
+                        platformProjectId = gce.ProjectId;
+                        platformResource = new MonitoredResource
+                        {
+                            Type = "gce_instance",
+                            Labels =
+                            {
+                                { "project_id", gce.ProjectId },
+                                { "instance_id", gce.InstanceId },
+                                { "zone", gce.ZoneName }
+                            }
+                        };
+                        break;
+                    case PlatformType.Unknown:
+                        GaxPreconditions.CheckState(!string.IsNullOrEmpty(ProjectId), $"{nameof(ProjectId)} must be set on this platform.");
+                        platformProjectId = ProjectId;
+                        platformResource = new MonitoredResource
+                        {
+                            Type = "global",
+                            Labels =
+                            {
+                                {"project_id", ProjectId }
+                            }
+                        };
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Unrecognized platform detected: '{platform.Type}'");
+                }
+                _logName = new LogName(PreferPlatformResourceTypeDetection ? platformProjectId : (ProjectId ?? platformProjectId), LogId).ToString();
```

Comment:
If the project ID differs than the detected project id,  Stackdriver Logging won't accept the log.    For that reason,  I hope you set PreferPlatformResourceTypeDetection  to true.    And we don't want user to specify project id normally.   If they do, when project id is different,  raise a warning or error.